### DISCUSSION TITLE
Laravel has multiple databases in the migration folder.

### DISCUSSION THREAD

**Discussion by bhargavraviya** [laravel/framework#48029] [Ideas] [answered]
In the migration folder, only single databases were handled, like MySQL and other databases (AWS RDS), but now different architecture databases are used at the same time, so migration problems arise.

My idea is **database/migrations**, folders, or it helps to manage multi-database migrations and virtual structures.

**Comment by henzeb** [comment]
I think it's a good idea. I'd suggest that each connection has a folder. So if you have a connection 'production' and a connection 'static-data' for data that almost never changes, we'd have migrate check the following folders for migrations and executes them per connection. 

- databases/migrations/production
- database/migrations/static-data

Of course, the ones in the migrations root go to your default connection, for backward compatibility.

**Comment by JurianArie** [comment]
You can add migrations for multiple databases using https://laravel.com/docs/10.x/migrations#setting-the-migration-connection

**Reply by henzeb** [reply]
yes, but can we separate them in folders? I mean it's not really visible what migration goes where. Having them structured makes much more sense.

**Reply by bhargavraviya** [reply]
@JurianArie 

The above question is for two types of databases MySQL and MongoDB
```
database/migrations/mysql 
database/migrations/mongodb
```

it working like 
MySQL table in the migration log diff?
MongoDB table in the migration log diff?

how to manage this type of structure?

**Reply by JurianArie** [reply, marked as answer]
You can tell the migrate command to look at multiple paths: https://github.com/laravel/framework/blob/5a7f2b4742b3dc7ce43acc698f400a9395801c7b/src/Illuminate/Database/Console/Migrations/MigrateCommand.php#L28

**Reply by henzeb** [reply]
Yes, but it should do that automatically with the subfolders of those paths (or the default ones), which it is not doing right now.

**Reply by JurianArie** [reply]
@henzeb feel free to create a PR. I got it working by replacing `glob` with `allFiles` but I don't know if there are edge-cases. https://github.com/laravel/framework/blob/5a7f2b4742b3dc7ce43acc698f400a9395801c7b/src/Illuminate/Database/Migrations/Migrator.php#L557

`$this->files->allFiles($path)`
